src/compiler/binder.ts

```diff
@@ -144,7 +163,10 @@ module ts {
             let symbol: Symbol;
             if (name !== undefined) {
                 symbol = hasProperty(symbols, name) ? symbols[name] : (symbols[name] = createSymbol(0, name));
-                if (symbol.flags & excludes) {
+                
+                // Check for declarations node cannot be merged with. 
+                // Interfaces declarations cannot be merged with non-ambient class declarations, which isn't encoded in SymbolFlags.

```

Comment:
`Instantiated` is used to indicate that a property or parameter was created by the process of instantiating its containing type or signature. It is only used for symbols created in the checker. So you should not use it here.